File: Step05_Fragmant/app/src/main/java/com/example/step05_fragmant/MainActivity.java
package com.example.step05_fragmant;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;

import com.example.step05_fragmant.databinding.ActivityMainBinding;
import com.example.step05_fragmant.databinding.FragmentMyBinding;

public class MainActivity extends AppCompatActivity implements MyFragment.MyFragmentListener {
    //MyFragment에서 전달하는 메세지를 받기 위해 인터페이스 구현

    ActivityMainBinding binding;
    FragmentMyBinding fragBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        //view binding 객체
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        FragmentManager fm = getSupportFragmentManager();

        MyFragment fragment1 = (MyFragment)fm.findFragmentById(R.id.fragment1);
        MyFragment fragment2 = (MyFragment)fm.findFragmentById(R.id.fragment2);
        binding.resetBtn.setOnClickListener(view -> {
            fragment1.reset();
            fragment2.reset();
        });

        binding.moveBtn.setOnClickListener(view -> {
            Intent loginIntent = new Intent(this, SubActivity.class);
            startActivity(loginIntent);
        });
    }

    @Override
    public void sendMsg(String msg) {
        Toast.makeText(this, "MyFragment : "+msg, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/step07_sqllite2/MainActivity.kt
package com.example.step07_sqllite2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.widget.Toast
import com.example.step07_sqllite2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), OnClickListener {
    lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        binding.sendBtn.setOnClickListener(this)
    }

    override fun onClick(p0: View?) {
        val msg = binding.inputMsg.text
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }
}